parsl/executors/workqueue/executor.py

```diff
@@ -355,7 +356,9 @@ def submit(self, func, resource_specification, *args, **kwargs):
                 logger.error(message)
                 raise ExecutorError(self, message)
 
-            if not self.autolabel and not keys.issuperset(required_resource_types):
+            key_check = required_resource_types.intersection(keys)
+            required_keys_ok = len(key_check) == 0 or key_check == required_resource_types
```

Comment:
I've added a comment.

The previous code railed an error if:
    autolabel was not set
    and some resource requirements were specified
    but some required_resource_types were not specified
 
This meant if you specify `cpus`, you also had to specify `mem` and `disk` (because work queue requries that).

This PR introduces a new resource type, `priority`, which has different requirement logic:
it is fine to specify `priority` independently of `cpus`/`mem`/`disk`
(and a future PR will introduce a wall time resource type, which will also be independent of those three).

The modified test in this PR checks that either:
* you specified *none* of cpus/mem/disk,
* or you specified *all* of cpus/mem/disk
while also allowing you to specify other unrelated resource types (`priority` and `gpus`)

Reply:
Makes sense.